pkg/api/datasources.go

```diff
@@ -347,6 +347,18 @@ func validateJSONData(jsonData *simplejson.Json, cfg *setting.Cfg) error {
 			}
 		}
 	}
+
+	for key, value := range jsonData.Get("teamHeaders").MustMap() {
```

Comment:
we want to validate the headers, so they do not override headers that should be part of the requests originally